js/core/team.js

```diff
@@ -519,6 +533,53 @@ define(["dao", "globals", "core/player", "lib/bluebird", "lib/underscore", "util
         options.playoffs = options.playoffs !== undefined ? options.playoffs : false;
         options.sortBy = options.sortBy !== undefined ? options.sortBy : "";
 
+
+        /**
+         * check if field is in sortBy
+         */
+        checkSort = function (sortBy, field) {
+            var cond, rev;
+            rev = '-' + field;
+            cond = Array.isArray(sortBy) && (sortBy.indexOf(field) > -1 || sortBy.indexOf(rev) > -1);
+            cond = cond || (sortBy === field || sortBy === rev);
+            return cond;
+        };
+
+
+        /**
+         * Assure needed fields are available for sorting.
+         */
+        assureSort = function (field, type, addFields) {
+            var cond;
+            type = type || 'seasonAttrs';
+            addFields = addFields || [];
+
+            _.each(addFields, function (f) {
+                if (_.isArray(f)) {
+                    options[f[1]] = options[f[1]] || [];
+                    options[f[1]].push(f[0]);
+                } else {
+                    options[type].push(f);
+                }
+            });
+
+            cond = checkSort(options.sortBy, field);
+            if (cond) {
+                options[type].push(field);
+            }
+        };
+
+        if (options.sortBy.length > 0) {
+            assureSort('winp', null, ['won', 'lost']);
```

Comment:
What's the point of checking all these things every time `filter` is called, and what happens if the requested sort is not possible?